kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java

```diff
@@ -299,63 +299,86 @@ public static String resolveValue(String variable) throws KubernetesPluginExcept
         }
         return variable;
     }
-
+    
     /**
-     * Generate map by splitting keyValues.
+     * Generate array of string using a {@link BLangArrayLiteral}.
      *
-     * @param keyValues key value paris.
-     * @return Map of key values.
+     * @param expr Array literal.
+     * @return Convert string.
      */
-    public static Map<String, String> getMap(List<BLangRecordLiteral.BLangRecordKeyValue> keyValues)
-            throws KubernetesPluginException {
-        Map<String, String> map = new LinkedHashMap<>();
-        if (keyValues != null) {
-            for (BLangRecordLiteral.BLangRecordKeyValue keyValue : keyValues) {
-                map.put(keyValue.getKey().toString(), resolveValue(keyValue.getValue().toString()));
+    public static List<String> getArray(BLangExpression expr) throws KubernetesPluginException {
+        if (expr instanceof BLangArrayLiteral) {
+            BLangArrayLiteral array = (BLangArrayLiteral) expr;
+            List<String> scopeSet = new LinkedList<>();
+            for (ExpressionNode bLangExpression : array.getExpressions()) {
+                scopeSet.add(getStringValue((BLangExpression) bLangExpression));
             }
+            return scopeSet;
+        } else {
+            throw new KubernetesPluginException("unable to parse value: " + expr.toString());
         }
-        return map;
     }
     
     /**
-     * Parse build extension of @kubernetes:Deployment annotation.
+     * Get a map from a ballerina expression.
      *
-     * @param buildExtensionValue Fields of the buildExtension field.
-     * @return Build extension model.
-     * @throws KubernetesPluginException When an unknown extension is found.
+     * @param expr Ballerina record value.
+     * @return Map of key values.
+     * @throws KubernetesPluginException When the expression cannot be parsed.
      */
-    public static DeploymentBuildExtension parseBuildExtension(BLangExpression buildExtensionValue)
-            throws KubernetesPluginException {
-        if (buildExtensionValue instanceof BLangSimpleVarRef || buildExtensionValue instanceof BLangLiteral) {
-            if ("openshift".equals(getStringValue(buildExtensionValue))) {
-                return new OpenShiftBuildExtensionModel();
+    public static Map<String, String> getMap(BLangExpression expr) throws KubernetesPluginException {
+        if (expr instanceof BLangRecordLiteral) {
```

Comment:
Here without using 'instanceof' we can check the node kind. We can also invert the if condition and return the error soon.

Reply:
Fixed with https://github.com/ballerinax/kubernetes/pull/301/commits/66ec19a0fb6d97a36b6121d8c27adf89e2956468